metadata/dashboard/dashboard_metadata_test.go

```diff
@@ -11,13 +11,15 @@ import (
 	"testing"
 
 	"github.com/featureform/metadata"
+	"github.com/featureform/metadata/search"
 	"github.com/featureform/provider"
 	"github.com/gin-gonic/gin"
 	"github.com/stretchr/testify/assert"
 	"go.uber.org/zap"
 )
 
 func GetTestGinContext(mockRecorder *httptest.ResponseRecorder) *gin.Context {
+	SearchClient = search.SearchMock{}
```

Comment:
What is the purpose of putting seach client here?
By calling SearchMock here we will test if it gets called successfully, but we don't do any further tests on what is returned and if that's an expected value. Is that something that I should consider?

Reply:
In this file, the `TestPostTags` test  needs to have the SearchClient instantiated otherwise it fails. That test calls the `PostTags` function in `dashboard_metadata.go` and that function uses `SearchClient.Upsert` around line 1454. (Really we just need it to exist, we don't really care to test the SearchClient itself)   

![Screenshot 2024-01-17 at 12 21 02 PM](https://github.com/featureform/featureform/assets/34227093/0c99cc0a-35ac-4477-9a98-8f77a6ffa46a)